src/net.rs

```diff
@@ -177,6 +180,57 @@ pub trait HostResponse {
     fn finish(&mut self) -> Result<(), Error>;
 }
 
+/// Represents a physical port that can be used to interact with client devices.
+///
+/// This for example is used for a PA RoT to send messages to a AC RoT.
+///
+/// This trait provides a generic mechanism for sending and receiving requests
+/// from devices.
+pub trait DevicePort<'a> {
+    /// The send function takes a `header` and `msg` to send to a device.
+    /// The implementation of this function must send the header and msg
+    /// to the device specified by `dest`.
+    ///
+    /// This should block until the operation is complete.
+    ///
+    /// On success it should return the number of bytes sent.
```

Comment:
What's a caller expected to do with this information? Seems like it might make more sense to return a `&mut dyn io::Write` instead of accepting a slice (this means, conveniently, that you don't need to buffer the message you're planning to send).

Reply:
I looked at returning an `io::Write` and using that, but it uses a large amount of stack space that I was trying to avoid. It also seems overally complex. I would prefer to just send the data from this function.

Reply:
Hmm, I don't follow why it uses "a lot" of stack space? Can you quantify?

While it is simpler to just send the data this way, it requires the sender to buffer up the encoded message, which may require using a lot of memory to do so. Avoiding this is one of Manticore's design goals (which has required significant complexity elsewhere). If you'd like we can merge this as-is, but I will probably wind up having to refactor it later.

Reply:
Yeah, so the current libtock-rs implementation uses 0x800 bytes of stack. When I changed this to returning an `io::Write` the stack was exceeded (and generated a fault), if I bump it up to 0x1000 it works again.

I can get more concrete numbers if you would like.

I'm not sure I understand the advantage of using `io::Write`. It adds a lot of complexity and is there a real case where you can get away without buffering the data? What hardware interface are you imaging?

Reply:
Let's just leave it like this for now. I can't promise I'll keep it like this indefinitely, though I still think that returning a `usize` is wrong and it should just block.